### PR TITLE
Move code around and add style.css header checks

### LINKED ISSUE #15: Multiple text domain issue
Theme Name: `Phpkida`
Text Domain: `Phpkida`

Sniffer message:
`Mismatch text domain. Expected 'Phpkida' or 'hybrid-core' but got 'phpkida'.`

May be we should rework on this.
```
$text_domains = array(
    // Current theme text domain.
    $theme->get( 'TextDomain' ),
    // Frameworks.
    'hybrid-core',
);
```

Also, in this `If we have a long list of framework text domains people will get confused. We will need a way to detect if a framework is being used and only then include that text domain to the whitelist.` since both issues are related.

### LINKED ISSUE #22: Add style header and file checking
`process_upload()` in https://meta.svn.wordpress.org/sites/trunk/wordpress.org/public_html/wp-content/plugins/theme-directory/class-wporg-themes-upload.php is doing a bunch of checks on the style.css header which is not in the theme check. We should create a separate function that would be included in the theme check. It would be easier to build this check in the theme check plugin then in PHPCS.

We could also check that the text domain is set right.

Could include the checking of the style header.
https://github.com/WordPress/theme-check/blob/master/checks/style_needed.php
https://github.com/WordPress/theme-check/blob/master/checks/style_suggested.php
https://github.com/WordPress/theme-check/blob/master/checks/style_tags.php

### PR DESCRIPTION
**Pull request by grappler**
This is a big PR which does a lot of things.

- Add a way to check text domain of frameworks. fixes #15
- Does style.css header checks. fixes #22

I also removed some extra divs from the markup and the summary as it does not fit in if there are multiple checks.

### CODE REVIEW

**Comment by ernilambar** [comment]
I tried this PR. All looks good. But PHP compat check is not working. Can you please check it once? Also CSS Lint is not here in this PR. Should we fix after this PR is merged?

**Comment by grappler** [comment]
Ah, yes, I did not copy all of the code across. I found another bug and made a few more code changes.

As a side note I am thinking of switching to [stylelint](https://github.com/stylelint/stylelint) instead.

**Comment by grappler** [comment]
The WordPress API does not support fetching the License and License URI. I have opened a ticket for it. https://core.trac.wordpress.org/ticket/40034

File: inc/checks.php [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+			'message' => sprintf( __( 'The text domain "%1$s" must match the theme slug "%2$s".', 'ns-theme-check' ),
+				$theme->get( 'TextDomain' ),
+				$theme_slug
+			),
+			'severity' => 'error',
+		);
+	}
+
+	$registered_tags    = ns_theme_check_get_theme_tags();
+	$tags               = array_map( 'strtolower', $theme->get('Tags') );
```
Comment (ernilambar) [inline inc/checks.php]: Travis is failing due to this line. `Expected 1 spaces after opening bracket; 0 found`

**Comment by grappler** [comment]
@ernilambar Ok, it should be fixed now.

**Pull request merged**